x-pack/plugins/security_solution/public/common/components/event_details/event_fields_browser.tsx

```diff
@@ -6,29 +6,73 @@
 
 import { sortBy } from 'lodash';
 import { EuiInMemoryTable } from '@elastic/eui';
-import React, { useMemo } from 'react';
+import React, { useCallback, useMemo } from 'react';
+import { useDispatch } from 'react-redux';
+import { rgba } from 'polished';
+import styled from 'styled-components';
 
+import { timelineActions, timelineSelectors } from '../../../timelines/store/timeline';
 import { ColumnHeaderOptions } from '../../../timelines/store/timeline/model';
 import { BrowserFields, getAllFieldsByName } from '../../containers/source';
 import { TimelineEventsDetailsItem } from '../../../../common/search_strategy/timeline';
-import { OnUpdateColumns } from '../../../timelines/components/timeline/events';
+import { getColumnHeaders } from '../../../timelines/components/timeline/body/column_headers/helpers';
+import { timelineDefaults } from '../../../timelines/store/timeline/defaults';
 
 import { getColumns } from './columns';
 import { search } from './helpers';
+import { useDeepEqualSelector } from '../../hooks/use_selector';
 
 interface Props {
   browserFields: BrowserFields;
-  columnHeaders: ColumnHeaderOptions[];
   data: TimelineEventsDetailsItem[];
   eventId: string;
-  onUpdateColumns: OnUpdateColumns;
   timelineId: string;
-  toggleColumn: (column: ColumnHeaderOptions) => void;
 }
 
+const TableWrapper = styled.div`
+  display: flex;
+  flex: 1;
+  overflow: hidden;
+
+  > div {
+    display: flex;
+    flex-direction: column;
+    flex: 1;
+    overflow: hidden;
+
+    > .euiFlexGroup:first-of-type {
+      flex: 0;
+    }
+  }
+`;
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const StyledEuiInMemoryTable = styled(EuiInMemoryTable as any)`
+  flex: 1;
+  overflow: auto;
+
+  &::-webkit-scrollbar {
+    height: ${({ theme }) => theme.eui.euiScrollBar};
+    width: ${({ theme }) => theme.eui.euiScrollBar};
+  }
+
+  &::-webkit-scrollbar-thumb {
+    background-clip: content-box;
+    background-color: ${({ theme }) => rgba(theme.eui.euiColorDarkShade, 0.5)};
+    border: ${({ theme }) => theme.eui.euiScrollBarCorner} solid transparent;
+  }
+
+  &::-webkit-scrollbar-corner,
+  &::-webkit-scrollbar-track {
+    background-color: transparent;
+  }
+`;
+
 /** Renders a table view or JSON view of the `ECS` `data` */
 export const EventFieldsBrowser = React.memo<Props>(
-  ({ browserFields, columnHeaders, data, eventId, onUpdateColumns, timelineId, toggleColumn }) => {
+  ({ browserFields, data, eventId, timelineId }) => {
+    const dispatch = useDispatch();
+    const getTimeline = useMemo(() => timelineSelectors.getTimelineByIdSelector(), []);
```

Comment:
Reading more and more your PR, I agree with you we should get more generic selector, the same way that we did/fix it for sourcerer.